quantmsutils/mzml/mzml_statistics.py

```diff
@@ -29,210 +44,229 @@ def mzml_statistics(ctx, ms_path: str, id_only: bool = False) -> None:
     peaks data for MS level 2 spectra.
 
     """
-    file_columns = [
-        "SpectrumID",
-        "MSLevel",
-        "Charge",
-        "MS_peaks",
-        "Base_Peak_Intensity",
-        "Summed_Peak_Intensities",
-        "Retention_Time",
-        "Exp_Mass_To_Charge",
-        "AcquisitionDateTime",
-    ]
+    schema = pa.schema([
+        pa.field("SpectrumID", pa.string(), nullable=True),
+        pa.field("MSLevel", pa.float64(), nullable=True),
+        pa.field("Charge", pa.float64(), nullable=True),
+        pa.field("MS_peaks", pa.float64(), nullable=True),
+        pa.field("Base_Peak_Intensity", pa.float64(), nullable=True),
+        pa.field("Summed_Peak_Intensities", pa.float64(), nullable=True),
+        pa.field("Retention_Time", pa.float64(), nullable=True),
+        pa.field("Exp_Mass_To_Charge", pa.float64(), nullable=True),
+        pa.field("AcquisitionDateTime", pa.string(), nullable=True),
+    ])
 
-    def parse_mzml(file_name: str, file_columns: list, id_only: bool = False):
+    def batch_write_mzml(
+        file_name: str,
+        parquet_schema: pa.Schema,
+        output_path: str,
+        id_only: bool = False,
+        batch_size: int = 10000
+    ) -> Optional[str]:
         """
         Parse mzML file and return a pandas DataFrame with the information. If id_only is True, it will also save a csv.
         @param file_name: The file name of the mzML file
-        @param file_columns: The columns of the DataFrame
         @param id_only: If True, it will save a csv with the spectrum id, mz and intensity
         @return: A pandas DataFrame with the information of the mzML file
         """
-
-        info = []
-        psm_part_info = []
         exp = MSExperiment()
-        acquisition_datetime = exp.getDateTime().get()
         MzMLFile().load(file_name, exp)
```

Comment:
This will still load the whole mzML into memory. If you want to be really efficient you can use our streaming mode.

https://pyopenms.readthedocs.io/en/release_2.4.0/file_handling.html#mzml-files-as-streams

Reply:
@jpfeuffer Thanks for the suggestion, Improved in last commit.